src/geofabrics/dem.py

```diff
@@ -241,13 +244,71 @@ def add_tile(self, tile_points: numpy.ndarray, window_size: int, idw_power: int,
             f"generated tile is not aligned with the overall dense DEM. The DEM raster origin is {raster_origin} " + \
             f"instead of {self.raster_origin}"
 
-        # trim to only include cells within catchment
+        # trim to only include cells within catchment - and update the tile extents with this new tile
         tile = tile.rio.clip(self.catchment_geometry.catchment.geometry)
         self._tiles = rioxarray.merge.merge_arrays([self._tiles, tile], method=method)
+        self._update_extents(tile_extent)
 
         # ensure the dem will be recalculated as another tile has been added
         self._dem = None
 
+    def _update_extents(self, tile_extent: geopandas.GeoDataFrame):
+        """ Update the extents of all LiDAR tiles updated """
+
+        assert len(tile_extent) == 1, "The tile_extent is expected to be contained in one shape. Instead " + \
+            f"tile_extent: {tile_extent} is of length {len(tile_extent)}."
+
+        if tile_extent.geometry.area.sum() > 0:  # check polygon isn't empty
+            if self._extents is None:
+                self._extents = tile_extent
+            else:
+                self._extents = geopandas.GeoDataFrame(
+                    {'geometry': [shapely.ops.cascaded_union([self._extents.loc[0].geometry,
+                                                              tile_extent.loc[0].geometry])]},
+                    crs=self.catchment_geometry.crs['horizontal'])
+            self._extents = geopandas.clip(self.catchment_geometry.catchment, self._extents)
+
+    def filter_lidar_extents_for_holes(self):
+        """ Remove holes below a filter size within the extents """
+
+        if self.area_to_drop is None:
```

Comment:
Do you still check/remove small holes if you have a demand to drop area offshore?

Reply:
Yup :) - added a comment to make this clearer!